pkg/requester/ranking/features.go

```diff
@@ -56,12 +56,12 @@ func NewStoragesNodeRanker() *featureNodeRanker[model.StorageSourceType] {
 // - Rank 10: Node is supporting the type(s) the job is requiring.
 // - Rank 0: We don't have information on what the node supports.
 // - Rank -1: Node is not supporting a type the job is requiring.
-func (s *featureNodeRanker[Key]) rankNode(ctx context.Context, node model.NodeInfo, requiredKeys []Key) int {
+func (s *featureNodeRanker[Key]) rankNode(ctx context.Context, node model.NodeInfo, requiredKeys []Key) (int, string) {
```

Comment:
Only a minor point, but I think this _might_ be one of those cases where named return values provide some useful info.  If this were 

```go 
func (s *featureNodeRanker[Key]) rankNode(
    ctx context.Context, 
    node model.NodeInfo, 
    requiredKeys []Key) (rank int, reason string) {
```
even if they weren't used they would still be useful as documentation.